uberfire-security/uberfire-security-management/uberfire-security-management-api/src/main/java/org/uberfire/ext/security/management/api/AbstractEntityManager.java

```diff
@@ -0,0 +1,114 @@
+package org.uberfire.ext.security.management.api;
+
+import org.uberfire.ext.security.management.api.exception.GenericUsersSystemManagementException;
+
+import java.util.List;
+
+/**
+ * <p>Basic management API for security realm entities type of <code>T</code>.</p>
+ * 
+ * @since 0.8.0
+ */
+public interface AbstractEntityManager<T> {
+
+    /**
+     * Search entities.
+     * @param request The search request constraints.
+     * @return List of resulting entities from the search result. 
+     * @throws GenericUsersSystemManagementException
+     */
+    SearchResponse<T> search(SearchRequest request) throws GenericUsersSystemManagementException;
```

Comment:
I think just SecurityManagamentException would suffice as name. The javadocs can point out that it's a general purpose exception.